sdks/python/apache_beam/io/parquetio.py

```diff
@@ -0,0 +1,349 @@
+a Parquet file. Records that are of simple types will be mapped into
+corresponding Python types. The actual parquet file operations are done by
+pyarrow. Source splitting is supported at row group granularity.
+
+Additionally, this module provides a write ``PTransform`` ``WriteToParquet``
+that can be used to write a given ``PCollection`` of Python objects to a
+Parquet file.
+"""
+from __future__ import absolute_import
+
+from functools import partial
+
+import pyarrow as pa
+from pyarrow.parquet import ParquetFile
+from pyarrow.parquet import ParquetWriter
+
+from apache_beam.io import filebasedsink
+from apache_beam.io import filebasedsource
+from apache_beam.io.filesystem import CompressionTypes
+from apache_beam.io.iobase import RangeTracker
+from apache_beam.io.iobase import Read
+from apache_beam.io.iobase import Write
+from apache_beam.transforms import PTransform
+
+__all__ = ['ReadFromParquet', 'ReadAllFromParquet', 'WriteToParquet']
+
+
+class ReadFromParquet(PTransform):
+  """A :class:`~apache_beam.transforms.ptransform.PTransform` for reading
+     Parquet files."""
+
+  def __init__(self, file_pattern=None, min_bundle_size=0,
+               validate=True, columns=None):
+    """Initialize :class:`ReadFromParquet`.
+    """
+    super(ReadFromParquet, self).__init__()
+    self._source = _create_parquet_source(
+        file_pattern,
+        min_bundle_size,
+        validate=validate,
+        columns=columns
+    )
+
+  def expand(self, pvalue):
+    return pvalue.pipeline | Read(self._source)
+
+  def display_data(self):
+    return {'source_dd': self._source}
+
+
+class ReadAllFromParquet(PTransform):
+  """A ``PTransform`` for reading ``PCollection`` of Parquet files.
+
+   Uses source '_ParquetSource' to read a ``PCollection`` of Parquet files or
+   file patterns and produce a ``PCollection`` of Parquet records.
+  """
+
+  DEFAULT_DESIRED_BUNDLE_SIZE = 64 * 1024 * 1024  # 64MB
+
+  def __init__(self, min_bundle_size=0,
+               desired_bundle_size=DEFAULT_DESIRED_BUNDLE_SIZE,
+               columns=None,
+               label='ReadAllFiles'):
+    """Initializes ``ReadAllFromParquet``.
+
+    Args:
+      min_bundle_size: the minimum size in bytes, to be considered when
+                       splitting the input into bundles.
+      desired_bundle_size: the desired size in bytes, to be considered when
+                       splitting the input into bundles.
+      columns: list of columns that will be read from files. A column name
+                       may be a prefix of a nested field, e.g. 'a' will select
+                       'a.b', 'a.c', and 'a.d.e'
+    """
+    super(ReadAllFromParquet, self).__init__()
+    source_from_file = partial(
+        _create_parquet_source,
+        min_bundle_size=min_bundle_size,
+        columns=columns
+    )
+    self._read_all_files = filebasedsource.ReadAllFiles(
+        True, CompressionTypes.AUTO, desired_bundle_size, min_bundle_size,
```

Comment:
Looks like this `CompressionTypes` is nothing related to parquet format. I assume that `CompressionTypes.AUTO` is just fine here since a parquet file itself can be compressed in any `CompressionTypes` such as `a.parquet.gz` or `b.parquet.bz2`.

Reply:
It would probably be preferable to simply disallow the specification of a compression type, as that's almost certainly an error and compression is handled internally to the format.